### PR TITLE
BaSP-MR-111: Redux application on Trainers

### PR DESCRIPTION
**Pull request by JuanManuel-Blois**
Applied redux and its logic in the trainers section of the app

### CODE REVIEW

File: src/Components/Trainers/index.js [hunk 1]
```diff
@@ -3,59 +3,44 @@ import styles from './trainers.module.css';
...
-  const deleteTrainer = async () => {
-    try {
-      const responseTrainer = await fetch(
-        `${process.env.REACT_APP_API_URL}/api/trainers/${idDelete}`,
-        {
-          method: 'DELETE'
-        }
-      );
+  useEffect(() => {
+    //agregar if chequeando si hay o no trainers
```
Comment (Tilajc) [inline src/Components/Trainers/index.js]: you can delete this line
```suggestion
    
```

File: src/Redux/Trainers/actions.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  return {
+    type: PUT_TRAINERS_PENDING
+  };
+};
+
+export const putTrainersSuccess = (updateTrainer, id) => {
+  return {
+    type: PUT_TRAINERS_SUCCESS,
+    payload: updateTrainer,
+    id
```
Comment (Tilajc) [inline src/Redux/Trainers/actions.js]: that's not necesary
```suggestion

```

File: src/Redux/Trainers/reducer.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      };
+
+    case DEL_TRAINERS_PENDING:
+      return {
+        ...state,
+        isPending: true,
+        error: null
+      };
+    case DEL_TRAINERS_SUCCESS: {
+      const idDeletedTrainer = [action.payload.id];
```
Comment (Tilajc) [inline src/Redux/Trainers/reducer.js]: that action.payload.id is undefined
```suggestion
      const idDeletedTrainer = [action.payload._id];
```

File: src/Redux/Trainers/reducer.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      const currentTrainers = state.data.filter((trainers) => trainers.id !== idDeletedTrainer);
```
Comment (Tilajc) [inline src/Redux/Trainers/reducer.js]: that trainers.id is undefined too
```suggestion
      const currentTrainers = state.data.filter((trainers) => trainers._id !== idDeletedTrainer);
```

File: src/Redux/Trainers/reducer.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      return {
+        ...state,
+        data: [currentTrainers],
```
Comment (Tilajc) [inline src/Redux/Trainers/reducer.js]: that is an array, you don't need to do an array inside another array
```suggestion
        data: currentTrainers,
```

**Review by Tilajc — CHANGES_REQUESTED** [review]
you're almost done! good job!

**Review by Tilajc — APPROVED** [review]
Well done!

**Comment by LucianoStradiot** [comment]
[Implement Redux on the Trainers entity](https://trello.com/c/8tcrvwJM/111-implement-redux-on-the-trainers-entity)

**Pull request merged**